crates/forge_services/src/truncator.rs

```diff
@@ -0,0 +1,231 @@
+#![allow(dead_code)]
+/// Maximum character limit for truncation
+const MAX_LIMIT: usize = 40_000;
+
+/// Result of a truncation operation
+#[derive(Debug, Clone, PartialEq)]
+pub struct TruncationResult {
+    /// The actual content passed for truncation.
+    pub actual: String,
+    /// The prefix portion of the truncated content (if applicable)
+    pub prefix: Option<String>,
+    /// The suffix portion of the truncated content (if applicable)
+    pub suffix: Option<String>,
```

Comment:
```suggestion
    pub prefix: Option<Range>,
    /// The suffix portion of the truncated content (if applicable)
    pub suffix: Option<Range>,
```

Instead of returning the complete content we can return ranges.